incomfortclient/__init__.py

```diff
@@ -237,26 +282,28 @@ async def update(self) -> None:
         _LOGGER.debug("Heater(%s).status() = %s", self._serial_no, self._status)
 
     @property
-    def status(self) -> dict:
+    def status(self) -> dict[str, Any]:
         """Return the current state of the heater."""
         return self._status
 
     @property
-    def display_code(self) -> int:
+    def display_code(self) -> DiplayCode | None:
         """Return the display code, 'displ_code'."""
-        return self._data["displ_code"]
+        return self._display_code
 
     @property
-    def display_text(self) -> None | str:
-        """Return the display code as text rather than a code."""
-        code = self.display_code
-        code_map = FAULT_CODES if self.is_failed else DISPLAY_CODES
-        return code_map.get(code, f"unknown/other, code = '{code}'")
+    def display_text(self) -> str:
+        """Return the display or fault code as text label rather than a code."""
+        if self.is_failed:
+            fault_code = self._fault_code or FaultCode.UNKNOWN
+            return fault_code.name.lower()
+        display_code = self._display_code or DiplayCode.UNKNOWN
+        return display_code.name.lower()
 
     @property
-    def fault_code(self) -> None | int:
+    def fault_code(self) -> FaultCode | None:
```

Comment:
I think we can remove this property now and remove the underscore from the inner variable

Reply:
I'd like to keep the property and add it as a reason to the binary sensor (as an extra state property).

Reply:
I mean `self._fault_code` -> `self.fault_code`

Reply:
Yeah, saw this later